src/main.nr

```diff
@@ -1,14 +1,57 @@
+use dep::std;
+
 global PRIME: u32 = 16777619;
+global HASH_BASIS: u32 = 2166136261;
+
+fn u32_to_u8(num: u32) -> [u8; 4] {
```

Comment:
You can also convert the u32 to field and use `.to_be_bytes(4);`. It helps you to avoid writing the whole function youself